doc/api/readline.md

```diff
@@ -500,7 +537,330 @@ Returns the real position of the cursor in relation to the input
+  * `history` {string[]} Initial list of history lines. This option makes sense
+    only if `terminal` is set to `true` by the user or by an internal `output`
+    check, otherwise the history caching mechanism is not initialized at all.
+    **Default:** `[]`.
+  * `historySize` {number} Maximum number of history lines retained. To disable
+    the history set this value to `0`. This option makes sense only if
+    `terminal` is set to `true` by the user or by an internal `output` check,
+    otherwise the history caching mechanism is not initialized at all.
+    **Default:** `30`.
+  * `removeHistoryDuplicates` {boolean} If `true`, when a new input line added
+    to the history list duplicates an older one, this removes the older line
+    from the list. **Default:** `false`.
+  * `prompt` {string} The prompt string to use. **Default:** `'> '`.
+  * `crlfDelay` {number} If the delay between `\r` and `\n` exceeds
+    `crlfDelay` milliseconds, both `\r` and `\n` will be treated as separate
+    end-of-line input. `crlfDelay` will be coerced to a number no less than
+    `100`. It can be set to `Infinity`, in which case `\r` followed by `\n`
+    will always be considered a single newline (which may be reasonable for
+    [reading files][] with `\r\n` line delimiter). **Default:** `100`.
+  * `escapeCodeTimeout` {number} The duration `readlinePromises` will wait for a
+    character (when reading an ambiguous key sequence in milliseconds one that
+    can both form a complete key sequence using the input read so far and can
+    take additional input to complete a longer key sequence).
+    **Default:** `500`.
+  * `tabSize` {integer} The number of spaces a tab is equal to (minimum 1).
+    **Default:** `8`.
+* Returns: {readlinePromises.Interface}
+
+The `readlinePromises.createInterface()` method creates a new `readlinePromises.Interface`
+instance.
+
+```js
+const readlinePromises = require('readlinePromises');
+const rl = readlinePromises.createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+```
+
+Once the `readlinePromises.Interface` instance is created, the most common case
+is to listen for the `'line'` event:
+
+```js
+rl.on('line', (line) => {
+  console.log(`Received: ${line}`);
+});
+```
+
+If `terminal` is `true` for this instance then the `output` stream will get
+the best compatibility if it defines an `output.columns` property and emits
+a `'resize'` event on the `output` if or when the columns ever change
+([`process.stdout`][] does this automatically when it is a TTY).
+
+#### Use of the `completer` function
+
+The `completer` function takes the current line entered by the user
+as an argument, and returns an `Array` with 2 entries:
+
+* An `Array` with matching entries for the completion.
+* The substring that was used for the matching.
+
+For instance: `[[substr1, substr2, ...], originalsubstring]`.
+
+```js
+function completer(line) {
```

Comment:
Can the completer function also be async? What happens if it returns a promise? What happens if the promise rejects?

Reply:
Good catch, I've updated the documentation. If the promise rejects, it's the same behavior as with the callback version:

```js
this[kWriteToOutput](`Tab completion error: ${inspect(err)}`);
```